cpp/src/copying/contiguous_split.cu

```diff
@@ -1158,122 +1265,752 @@ std::vector<packed_table> contiguous_split(cudf::table_view const& input,
+    // return false, just in case
+    chunk_iter_state->advance_iteration();
+
+    return make_packed_tables();
+  }
+
+  cudf::size_type contiguous_split_chunk(cudf::device_span<uint8_t> const& user_buffer)
+  {
+    CUDF_FUNC_RANGE()
+    CUDF_EXPECTS(
+      user_buffer.size() == user_buffer_size,
+      "Cannot use a device span smaller than the output buffer size configured at instantiation!");
+    CUDF_EXPECTS(has_next(), "Cannot call contiguous_split_chunk with has_next() == false!");
+
+    auto [starting_batch, num_batches_to_copy] =
+      chunk_iter_state->get_current_starting_index_and_buff_count();
+
+    // perform the copy.
+    copy_data(num_batches_to_copy,
+              starting_batch,
+              src_and_dst_pointers->d_src_bufs,
+              src_and_dst_pointers->d_dst_bufs,
+              chunk_iter_state->d_batched_dst_buf_info,
+              user_buffer.data(),
+              stream);
+
+    // We do not need to post-process null counts since the null count info is
+    // taken from the source table in the contiguous_split_chunk case (no splits)
+    return chunk_iter_state->advance_iteration();
+  }
+
+  std::vector<packed_table> make_empty_packed_table()
+  {
+    // sanitize the inputs (to handle corner cases like sliced tables)
+    std::vector<cudf::column_view> empty_column_views;
+    empty_column_views.reserve(input.num_columns());
+    std::transform(input.begin(),
+                   input.end(),
+                   std::back_inserter(empty_column_views),
+                   [](column_view const& col) { return cudf::empty_like(col)->view(); });
+
+    table_view empty_inputs(empty_column_views);
+
+    // build the empty results
+    std::vector<packed_table> result;
+    result.reserve(num_partitions);
+    auto const iter = thrust::make_counting_iterator(0);
+    std::transform(iter,
+                   iter + num_partitions,
+                   std::back_inserter(result),
+                   [&empty_inputs](int partition_index) {
+                     return packed_table{
+                       empty_inputs,
+                       packed_columns{std::make_unique<std::vector<uint8_t>>(pack_metadata(
+                                        empty_inputs, static_cast<uint8_t const*>(nullptr), 0)),
+                                      std::make_unique<rmm::device_buffer>()}};
+                   });
+
+    return result;
+  }
+
+  std::unique_ptr<std::vector<uint8_t>> build_packed_column_metadata()
+  {
+    CUDF_EXPECTS(num_partitions == 1, "build_packed_column_metadata supported only without splits");
+
+    if (input.num_columns() == 0) { return std::unique_ptr<std::vector<uint8_t>>(); }
+
+    if (is_empty) {
+      // this is a bit ugly, but it was done to re-use make_empty_packed_table between the
+      // regular contiguous_split and chunked_pack cases.
+      auto empty_packed_tables = std::move(make_empty_packed_table()[0]);
```

Comment:
```suggestion
      auto empty_packed_tables = std::move(make_empty_packed_table().front());
```

Reply:
Done https://github.com/rapidsai/cudf/pull/13260/commits/ab98d70bce02b59dab1f38a8da8162a999a70167 (and the next one included there)